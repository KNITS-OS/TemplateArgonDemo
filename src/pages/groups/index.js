export * from "./create-group";
export * from "./group-details";
export * from "./group-panels";
export * from "./search-groups";

// has to be the last export
export * from "./groups.routes.const";
export * from "./groups.routes";
