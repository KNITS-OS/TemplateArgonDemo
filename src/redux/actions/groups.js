import { fetchGroupData } from "redux/services/queries";
import { API_CALL_ERROR, API_CALL_START, SEARCH_GROUPS } from "./types";

// export const createGroup = (name, description) => async dispatch => {
//   try {
//     const res = await groupServices.create({ name, description });
//     dispatch({
//       type: CREATE_GROUP,
//       payload: res.data,
//     });
//     return Promise.resolve(res.data);
//   } catch (err) {
//     console.log(err);
//     return Promise.reject(err);
//   }
// };

// export const retrieveGroups = () => async dispatch => {
//   try {
//     dispatch({
//       type: API_CALL_START,
//       payload: RETRIEVE_GROUPS,
//     });

//     const { data } = await groupServices.getAll("*");

//     dispatch({
//       type: RETRIEVE_GROUPS,
//       payload: data,
//     });
//   } catch (err) {
//     dispatch({
//       type: API_CALL_ERROR,
//       payload: err,
//     });
//   }
// };

export const searchGroups = () => async dispatch => {
  try {
    dispatch({
      type: API_CALL_START,
      payload: SEARCH_GROUPS,
    });
    // search
    let { data } = await fetchGroupData();

    dispatch({
      type: SEARCH_GROUPS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: API_CALL_ERROR,
      payload: err,
    });
  }
};

// export const addGroupMember = (id, data) => async dispatch => {
//   try {
//     const res = await groupServices.update(id, data);
//     dispatch({
//       type: ADD_CAREMEMBER_TO_GROUP,
//       payload: data,
//     });

//     return Promise.resolve(res.data);
//   } catch (err) {
//     return Promise.reject(err);
//   }
// };

// export const deactivateGroup = id => async dispatch => {
//   try {
//     await groupServices.update(id);
//     dispatch({
//       type: DEACTIVATE_GROUP,
//       payload: { id },
//     });
//   } catch (err) {
//     console.log(err);
//   }
// };

// export const removeGroupMember = (id, data) => async dispatch => {
//   try {
//     await groupServices.update(id, data);

//     dispatch({
//       type: REMOVE_CAREMEMBER_FROM_GROUP,
//       payload: { id },
//     });
//   } catch (err) {
//     console.log(err);
//   }
// };
