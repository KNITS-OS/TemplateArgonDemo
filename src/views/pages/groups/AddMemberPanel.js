/*!

=========================================================
* Argon Dashboard PRO React - v1.2.0
=========================================================

* Product Page: https://www.creative-tim.com/product/argon-dashboard
* Copyright 2021 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React from "react";
import { useSelector } from "react-redux";
// core components
import Select from "react-select";
// reactstrap components
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  FormGroup,
  Row,
} from "reactstrap";

function AddMemberPanel(props) {
  const employeesData = useSelector(state => state.employees);

  const employees = employeesData.map(employee => {
    return { value: employee.id, label: employee.internationalName };
  });

  const getBusinessUnits = useSelector(state => {
    return state.categories.businessUnits.map(bunit => {
      return { value: bunit.id, label: bunit.name };
    });
  });

  const getCountries = useSelector(state => {
    return state.categories.countryListAllIsoData.map(country => {
      return { value: country.code3, label: country.name };
    });
  });

  const jobTitles = [
    { value: 1, label: "product manager" },
    { value: 2, label: "qa engineer" },
    { value: 3, label: "hr consultant" },
    { value: 4, label: "office manager" },
    { value: 5, label: "sales representative" },
    { value: 6, label: "logistics consultant" },
  ];

  return (
    <Card>
      <CardHeader>
        <Row className="align-items-center">
          <Col xs="8">
            <h3 className="mb-0">Add Members</h3>
          </Col>
        </Row>
      </CardHeader>
      <CardBody>
        <Row>
          <Col xl="12">
            <Row>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Job Title
                  </label>
                  <Select
                    onChange={props.onchangeRole}
                    options={jobTitles}
                    // getOptionValue={(option) => option.value}
                    // getOptionLabel={(option) => option.value}
                  />
                </FormGroup>
              </Col>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Country
                  </label>
                  <Select
                    onChange={props.onchangeCountry}
                    options={getCountries}
                    // getOptionValue={(option) => option.value}
                    // getOptionLabel={(option) => option.name}
                  />
                </FormGroup>
              </Col>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Business Unit
                  </label>
                  <Select
                    onChange={props.onchangeBunit}
                    options={getBusinessUnits}
                  />
                </FormGroup>
              </Col>
              <Col sm="6">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Add members
                  </label>
                  <Select
                    isMulti
                    onChange={props.onSelectCareMember}
                    options={employees}
                  />
                </FormGroup>
              </Col>
            </Row>
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
}

export default AddMemberPanel;

// AddMemberPanel.propTypes = {
//   onchangeRole: PropTypes.func,
//   onchangeCountry: PropTypes.func,
//   onchangeBunit: PropTypes.func,
//   onSelectCareMember: PropTypes.func
// };
