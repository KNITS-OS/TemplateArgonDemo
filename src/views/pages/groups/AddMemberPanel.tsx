/*!

=========================================================
* Argon Dashboard PRO React - v1.2.0
=========================================================

* Product Page: https://www.creative-tim.com/product/argon-dashboard
* Copyright 2021 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
// core components
import Select, { MultiValue, SingleValue } from "react-select";
// reactstrap components
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  FormGroup,
  Row,
} from "reactstrap";
import { OptionType } from "types/types";
import {
  getSelectBusinessUnits,
  getSelectCountries,
  getSelectEmployees,
} from "utils/fetchData";

interface Props {
  onChangeRole: (option: SingleValue<OptionType>) => void;
  onChangeCountry: (option: SingleValue<OptionType>) => void;
  onChangeBunit: (option: SingleValue<OptionType>) => void;
  onSelectCareMember: (option: MultiValue<OptionType>) => void;
}

const AddMemberPanel = ({
  onChangeRole,
  onChangeCountry,
  onChangeBunit,
  onSelectCareMember,
}: Props) => {
  const jobTitles: OptionType[] = [
    { value: "1", label: "product manager" },
    { value: "2", label: "qa engineer" },
    { value: "3", label: "hr consultant" },
    { value: "4", label: "office manager" },
    { value: "5", label: "sales representative" },
    { value: "6", label: "logistics consultant" },
  ];

  return (
    <Card>
      <CardHeader>
        <Row className="align-items-center">
          <Col xs="8">
            <h3 className="mb-0">Add Members</h3>
          </Col>
        </Row>
      </CardHeader>
      <CardBody>
        <Row>
          <Col xl="12">
            <Row>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Job Title
                  </label>
                  <Select
                    onChange={item => onChangeRole(item)}
                    options={jobTitles}
                    getOptionValue={option => option.value}
                    getOptionLabel={option => option.label}
                  />
                </FormGroup>
              </Col>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Country
                  </label>
                  <Select
                    onChange={item => onChangeCountry(item)}
                    options={getSelectCountries}
                    getOptionValue={option => option.value}
                    getOptionLabel={option => option.label}
                  />
                </FormGroup>
              </Col>
              <Col xl="2">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Business Unit
                  </label>
                  <Select
                    onChange={item => onChangeBunit(item)}
                    options={getSelectBusinessUnits}
                    getOptionValue={option => option.value}
                    getOptionLabel={option => option.label}
                  />
                </FormGroup>
              </Col>
              <Col sm="6">
                <FormGroup>
                  <label className="form-control-label" htmlFor="members">
                    Add members
                  </label>
                  <Select
                    onChange={item => onSelectCareMember(item)}
                    options={getSelectEmployees}
                    isMulti
                    getOptionValue={option => option.value}
                    getOptionLabel={option => option.label}
                  />
                </FormGroup>
              </Col>
            </Row>
          </Col>
        </Row>
      </CardBody>
    </Card>
  );
};

export default AddMemberPanel;
