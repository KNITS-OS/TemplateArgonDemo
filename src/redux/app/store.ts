import { configureStore } from "@reduxjs/toolkit";
import counterReducer from "redux/features/example/counterSlice";
import sidenavReducer from "redux/features/sidenav/sidenavSlice";
import { employeesApiSlice } from "../features/employees/employeesApiSlice";
import { groupsApiSlice } from "../features/groups/groupsApiSlice";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    sidenav: sidenavReducer,
    [employeesApiSlice.reducerPath]: employeesApiSlice.reducer,
    [groupsApiSlice.reducerPath]: groupsApiSlice.reducer,
  },
  // add special capabilites to the store
  // like tracking cache lifetimes
  // if no other part of the codebase needs this data
  // it will be removed from the cache after it expires
  middleware: getDefaultMiddleware => {
    return getDefaultMiddleware().concat(
      employeesApiSlice.middleware,
      groupsApiSlice.middleware,
    );
  },
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDisptatch = typeof store.dispatch;
