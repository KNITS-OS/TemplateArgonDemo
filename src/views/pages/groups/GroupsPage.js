/*!

=========================================================
* Argon Dashboard PRO React - v1.2.0
=========================================================

* Product Page: https://www.creative-tim.com/product/argon-dashboard-react
* Copyright 2021 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
// core components
import SimpleHeader from "components/Headers/SimpleHeader.js";
import TableHeader from "components/Headers/TableHeader";
import { groups } from "mock-data/groups.js";
import React from "react";
// react component for creating dynamic tables
import BootstrapTable from "react-bootstrap-table-next";
import ToolkitProvider, { Search } from "react-bootstrap-table2-toolkit";
import { useHistory } from "react-router";
// reactstrap components
import { Button, Card, Container, Row } from "reactstrap";
import { pagination } from "utils/tableUtils";

const { SearchBar } = Search;

function GroupsPage() {
  const history = useHistory();

  const groupDetails = e => {
    var { id } = e.target;
    history.push(`/admin/groups/group-details/${id}`);
  };

  const formatActionButtonCell = (cell, row) => {
    return (
      <>
        <Button
          id={row.id}
          className="btn-icon btn-2"
          type="button"
          color="info"
          onClick={groupDetails}
        >
          <span id={row.id} className="btn-inner--icon">
            <i id={row.id} className="ni ni-badge" />
          </span>
        </Button>
      </>
    );
  };

  return (
    <>
      {alert}
      <SimpleHeader name="React Tables" parentName="Tables" />
      <Container className="mt--6" fluid>
        <Row>
          <div className="col">
            <Card>
              <TableHeader
                title="React Bootstrap Table 2"
                text="This is an exmaple of data table using the well known
                  react-bootstrap-table2 plugin. This is a minimal setup in
                  order to get started fast."
              />

              <ToolkitProvider
                data={groups}
                keyField="id"
                columns={[
                  {
                    dataField: "id",
                    text: "id",
                    hidden: true,
                  },
                  {
                    dataField: "name",
                    text: "Name",
                    sort: true,
                  },
                  {
                    dataField: "description",
                    text: "description",
                    sort: true,
                  },
                  {
                    dataField: "active",
                    text: "active",
                    sort: true,
                  },
                  {
                    dataField: "action",
                    text: "",
                    formatter: formatActionButtonCell,
                  },
                ]}
                search
              >
                {props => (
                  <div className="py-4 table-responsive">
                    <div
                      id="datatable-basic_filter"
                      className="dataTables_filter px-4 pb-1"
                    >
                      <label>
                        Search:
                        <SearchBar
                          className="form-control-sm"
                          {...props.searchProps}
                        />
                      </label>
                    </div>
                    <BootstrapTable
                      {...props.baseProps}
                      bootstrap4={true}
                      pagination={pagination}
                      bordered={false}
                    />
                  </div>
                )}
              </ToolkitProvider>
            </Card>
          </div>
        </Row>
      </Container>
    </>
  );
}

export default GroupsPage;
