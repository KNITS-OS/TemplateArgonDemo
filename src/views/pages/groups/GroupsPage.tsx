/*!

=========================================================
* Argon Dashboard PRO React - v1.2.0
=========================================================

* Product Page: https://www.creative-tim.com/product/argon-dashboard-react
* Copyright 2021 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
// core components
import SimpleHeader from "components/Headers/SimpleHeader";
// react component for creating dynamic tables
import BootstrapTable from "react-bootstrap-table-next";
import ToolkitProvider, { Search } from "react-bootstrap-table2-toolkit";
import { useHistory } from "react-router";
// reactstrap components
import {
  Button,
  Card,
  CardHeader,
  Col,
  Container,
  FormGroup,
  Row,
} from "reactstrap";
import { Group } from "types/types";
import { pagination } from "utils/tableUtils";
import { useAppDispatch, useAppSelector } from "../../../redux/app/hooks";
import { fetchGroups } from "../../../redux/features/groups/groupsSlice";

const { SearchBar } = Search;

const GroupsPage = () => {
  const history = useHistory();

  const groupDetails = (e: React.MouseEvent<HTMLButtonElement>) => {
    var { id } = e.target as HTMLButtonElement;
    history.push(`/admin/groups/group-details/${id}`);
  };

  const { groups = [] } = useAppSelector(state => state.groups);
  const dispatch = useAppDispatch();

  const formatActionButtonCell = (_: undefined, row: Group) => {
    const { id } = row;

    let groupId = id.toString();
    return (
      <>
        <Button
          id={groupId}
          className="btn-icon btn-2"
          type="button"
          color="info"
          onClick={groupDetails}
        >
          <span id={groupId} className="btn-inner--icon">
            <i id={groupId} className="ni ni-badge" />
          </span>
        </Button>
      </>
    );
  };

  const onFindGroups = () => {
    dispatch(fetchGroups());
  };

  return (
    <>
      {alert}
      <SimpleHeader name="React Tables" parentName="Tables" />
      <Container className="mt--6" fluid>
        <Row>
          <div className="col">
            <Card>
              <CardHeader>
                <h3 className="mb-0">React Bootstrap Table 2</h3>
                <p className="text-sm mb-0">
                  This is an exmaple of data table using the well known
                  react-bootstrap-table2 plugin. This is a minimal setup in
                  order to get started fast.
                </p>
              </CardHeader>
              <ToolkitProvider
                data={groups}
                keyField="id"
                columns={[
                  {
                    dataField: "id",
                    text: "id",
                    hidden: true,
                  },
                  {
                    dataField: "name",
                    text: "Name",
                    sort: true,
                  },
                  {
                    dataField: "description",
                    text: "description",
                    sort: true,
                  },
                  {
                    dataField: "active",
                    text: "active",
                    sort: true,
                  },
                  {
                    dataField: "action",
                    text: "",
                    formatter: formatActionButtonCell,
                  },
                ]}
                search
              >
                {props => (
                  <div className="py-4 table-responsive">
                    <div
                      id="datatable-basic_filter"
                      className="dataTables_filter px-4 pb-1"
                    >
                      <Row>
                        <Col>
                          <label>
                            Search:
                            <SearchBar
                              className="form-control-sm"
                              {...props.searchProps}
                            />
                          </label>
                        </Col>
                        <Col>
                          <FormGroup>
                            <button
                              className="btn btn-info"
                              type="button"
                              onClick={onFindGroups}
                            >
                              Search
                            </button>
                          </FormGroup>
                        </Col>
                      </Row>
                    </div>
                    <BootstrapTable
                      {...props.baseProps}
                      bootstrap4={true}
                      pagination={pagination}
                      bordered={false}
                    />
                  </div>
                )}
              </ToolkitProvider>
            </Card>
          </div>
        </Row>
      </Container>
    </>
  );
};

export default GroupsPage;
