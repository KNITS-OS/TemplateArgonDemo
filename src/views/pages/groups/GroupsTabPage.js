import React, { useState } from "react";
import { Card, CardBody, Nav, TabContent, TabPane } from "reactstrap";
import NavItemTabButton from "views/components/NavItemTabButton";
import CreateGroupPage from "./CreateGroupPage";
import GroupsPage from "./GroupsPage";

const GroupsTabPage = () => {
  const tabId1 = "Search Groups";
  const tabId2 = "Create Group";
  const [tab, setTab] = useState(tabId1);

  return (
    <Card>
      <CardBody>
        <Nav pills style={{ marginBottom: 12 }}>
          <NavItemTabButton tabId={tabId1} tab={tab} setTab={setTab} />
          <NavItemTabButton tabId={tabId2} tab={tab} setTab={setTab} />
        </Nav>
        <TabContent activeTab={tab}>
          <TabPane tabId={tabId1}>
            <GroupsPage />
          </TabPane>
          <TabPane tabId={tabId2}>
            <CreateGroupPage />
          </TabPane>
        </TabContent>
      </CardBody>
    </Card>
  );
};
export default GroupsTabPage;
