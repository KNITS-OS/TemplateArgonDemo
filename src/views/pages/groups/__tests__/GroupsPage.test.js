import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { searchGroups } from "redux/actions/groups";
import { fetchGroupData } from "redux/services/queries";
import store from "store";
import GroupsPage from "../GroupsPage";

const MockGroupsPage = () => {
  return (
    <Provider store={store}>
      <GroupsPage />
    </Provider>
  );
};

describe("Groups Page", () => {
  describe("Table", () => {
    beforeEach(() => {
      jest.mock("../../../../__mocks__/axios");
    });

    test("should render multible table items", async () => {
      render(<MockGroupsPage />);

      store.dispatch(searchGroups());

      const text = await screen.findByText("Brainy Buddies");
      expect(text).toBeInTheDocument();
    });

    test("should get groups data", async () => {
      const { data } = await fetchGroupData();

      expect(data).toHaveLength(5);
    });

    test("should click on search and render groups", async () => {
      render(<MockGroupsPage />);
      const button = screen.getByRole("button", {
        name: /search/i,
      });
      userEvent.click(button);

      const text = await screen.findByText("Brainy Buddies");

      expect(text).toBeInTheDocument();
    });
  });
});
