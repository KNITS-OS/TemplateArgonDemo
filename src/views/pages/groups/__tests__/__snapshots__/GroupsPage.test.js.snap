// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups Page Snapshot tests should render correctly 1`] = `
<DocumentFragment>
  <div
    class="header header-dark bg-info pb-6 content__title content__title--calendar"
  >
    <div
      class="container-fluid"
    >
      <div
        class="header-body"
      >
        <div
          class="align-items-center py-4 row"
        >
          <div
            class="col-7 col-lg-6"
          >
            <h6
              class="fullcalendar-title h2 text-white d-inline-block mb-0"
            >
              React Tables
            </h6>
             
            <nav
              aria-label="breadcrumb"
              class="d-none d-md-inline-block ml-lg-4"
            >
              <ol
                class="breadcrumb breadcrumb-links breadcrumb-dark"
              >
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="#pablo"
                  >
                    <i
                      class="fas fa-home"
                    />
                  </a>
                </li>
                <li
                  class="breadcrumb-item"
                >
                  <a
                    href="#pablo"
                  >
                    Tables
                  </a>
                </li>
                <li
                  class="active breadcrumb-item"
                >
                  React Tables
                </li>
              </ol>
            </nav>
          </div>
          <div
            class="mt-3 mt-md-0 text-md-right col-5 col-lg-6"
          >
            <button
              class="btn-neutral btn btn-default btn-sm"
            >
              New
            </button>
            <button
              class="btn-neutral btn btn-default btn-sm"
            >
              Filters
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mt--6 container-fluid"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            <h3
              class="mb-0"
            >
              React Bootstrap Table 2
            </h3>
            <p
              class="text-sm mb-0"
            >
              This is an exmaple of data table using the well known
 react-bootstrap-table2 plugin. This is a minimal setup in
 order to get started fast.
            </p>
          </div>
          <div
            class="py-4 table-responsive"
          >
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="dataTables_filter px-4 pb-1"
                  id="datatable-basic_filter"
                >
                  <label>
                    Search:
                    <label
                      class="search-label"
                      for="search-bar-0"
                    >
                      <span
                        class="sr-only"
                        id="search-bar-0-label"
                      >
                        Search this table
                      </span>
                      <input
                        aria-labelledby="search-bar-0-label"
                        class="form-control form-control-sm"
                        id="search-bar-0"
                        placeholder="Search"
                        type="text"
                        value=""
                      />
                    </label>
                  </label>
                </div>
              </div>
              <div
                class="col-md-2"
                style="display: flex; justify-content: center;"
              >
                <button
                  class="btn btn-info"
                >
                  Search
                </button>
              </div>
            </div>
            <div
              class="react-bootstrap-table"
            >
              <table
                class="table"
              >
                <thead>
                  <tr>
                    <th
                      aria-label="Name sortable"
                      class="sortable"
                      tabindex="0"
                    >
                      Name
                      <span
                        class="order-4"
                      />
                    </th>
                    <th
                      aria-label="description sortable"
                      class="sortable"
                      tabindex="0"
                    >
                      description
                      <span
                        class="order-4"
                      />
                    </th>
                    <th
                      aria-label="active sortable"
                      class="sortable"
                      tabindex="0"
                    >
                      active
                      <span
                        class="order-4"
                      />
                    </th>
                    <th
                      tabindex="0"
                    />
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Brainy Buddies
                    </td>
                    <td>
                      Nothing can be achieved without having fun.
                    </td>
                    <td>
                      true
                    </td>
                    <td>
                      <button
                        class="btn-icon btn-2 btn btn-info"
                        id="1"
                        type="button"
                      >
                        <span
                          class="btn-inner--icon"
                          id="1"
                        >
                          <i
                            class="ni ni-badge"
                            id="1"
                          />
                        </span>
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Dynamite Dealers
                    </td>
                    <td>
                      Can't beat them at negotiation or Monopoly..
                    </td>
                    <td>
                      true
                    </td>
                    <td>
                      <button
                        class="btn-icon btn-2 btn btn-info"
                        id="2"
                        type="button"
                      >
                        <span
                          class="btn-inner--icon"
                          id="2"
                        >
                          <i
                            class="ni ni-badge"
                            id="2"
                          />
                        </span>
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Unfrozen Caveman
                    </td>
                    <td>
                      If you need extreme solution they are there for you. Might be rude from time to time
                    </td>
                    <td>
                      true
                    </td>
                    <td>
                      <button
                        class="btn-icon btn-2 btn btn-info"
                        id="3"
                        type="button"
                      >
                        <span
                          class="btn-inner--icon"
                          id="3"
                        >
                          <i
                            class="ni ni-badge"
                            id="3"
                          />
                        </span>
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Mind Serenade
                    </td>
                    <td>
                      They always comes with new ideas, out from the box
                    </td>
                    <td>
                      true
                    </td>
                    <td>
                      <button
                        class="btn-icon btn-2 btn btn-info"
                        id="4"
                        type="button"
                      >
                        <span
                          class="btn-inner--icon"
                          id="4"
                        >
                          <i
                            class="ni ni-badge"
                            id="4"
                          />
                        </span>
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Clan Optimism
                    </td>
                    <td>
                      Watch the bright side of everything
                    </td>
                    <td>
                      true
                    </td>
                    <td>
                      <button
                        class="btn-icon btn-2 btn btn-info"
                        id="5"
                        type="button"
                      >
                        <span
                          class="btn-inner--icon"
                          id="5"
                        >
                          <i
                            class="ni ni-badge"
                            id="5"
                          />
                        </span>
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="row react-bootstrap-table-pagination"
            >
              <div
                class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
              >
                <div
                  class="dataTables_length"
                  id="datatable-basic_length"
                >
                  <label>
                    Show 
                    <select
                      aria-controls="datatable-basic"
                      class="form-control form-control-sm"
                      name="datatable-basic_length"
                    >
                      <option
                        value="10"
                      >
                        10
                      </option>
                      <option
                        value="25"
                      >
                        25
                      </option>
                      <option
                        value="50"
                      >
                        50
                      </option>
                      <option
                        value="100"
                      >
                        100
                      </option>
                    </select>
                     entries.
                  </label>
                </div>
                <span
                  class="react-bootstrap-table-pagination-total"
                >
                   Showing rows 1 to 5 of 5
                </span>
              </div>
              <div
                class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
              >
                <ul
                  class="pagination react-bootstrap-table-page-btns-ul"
                >
                  <li
                    class="disabled page-item"
                    title="previous page"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      &lt;
                    </a>
                  </li>
                  <li
                    class="active page-item"
                    title="1"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      1
                    </a>
                  </li>
                  <li
                    class="disabled page-item"
                    title="next page"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      &gt;
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
